Core/include/Acts/Propagator/Navigator.hpp

```diff
@@ -706,8 +706,7 @@ class Navigator {
 
           // Check: are we on the first surface?
           if (state.navigation.currentSurface == nullptr ||
-              state.navigation.currentSurface !=
-                  protoNavSurfaces.front().object) {
+              protoNavSurfaces.front().intersection.pathLength > 1_um) {
```

Comment:
Is this concerning edge-cases?

Reply:
Yes, this is only for navigation in calorimeter like detector (@paulgessinger can confirm but I think right now this is only use in the ATLAS Calorimeter). It resolve an edge case when two surface are at the same position which result in an extrapolation from that surface to itself forever.

Reply:
Yeah I think this branch only becomes active if the tracking volume has BVH with e.g. calo cells.